hazelcast/src/main/java/com/hazelcast/map/impl/query/QueryResult.java

```diff
@@ -62,7 +62,7 @@
  */
 public class QueryResult implements Result<QueryResult>, Iterable<QueryResultRow> {
 
-    private List rows = new LinkedList();
+    private List rows = new ArrayList();
```

Comment:
`QueryResult` constructor has `resultLimit`, it can be better to pass it to `ArrayList` as an initial capacity. Seems initial capacity can be applied also for `QueryResult#readData`.